src/power_grid_model/core/power_grid_dataset.py

```diff
@@ -266,6 +273,18 @@ def _validate_properties(self, data: ComponentData, schema: ComponentMetaData):
         if properties.batch_size != self._batch_size:
             raise ValueError(f"Dataset must have a consistent batch size across all components. {VALIDATOR_MSG}")
 
+    def _validate_dtypes_compatibility(self, data: ComponentData, schema: ComponentMetaData):
+        sub_data = _extract_data_from_component_data(data)
+        if is_columnar(data):
+            for attr, array in sub_data.items():
+                if schema.dtype.names is None or attr not in schema.dtype.names:
+                    raise Warning(f"Attribute {attr} is not in schema. {VALIDATOR_MSG}")
+                if array.dtype != schema.dtype[attr]:
```

Comment:
And this would fix the tests
```suggestion
                if array.dtype != schema.dtype[attr].base:
```

Reply:
as i mentioned in the other comment: this + shape check is the best we can do here